delocate/libsana.py

```diff
@@ -4,16 +4,198 @@
+            if install_name.startswith("@"):
+                dependency_path = resolve_rpath(
+                    install_name,
+                    rpaths,
+                    loader_path=dirname(lib_fname),
+                )
+            else:
+                dependency_path = search_environment_for_lib(install_name)
+            yield dependency_path, install_name
+            if dependency_path != install_name:
+                logger.debug(
+                    "%s resolved to: %s", install_name, dependency_path
+                )
+        except DependencyNotFound:
+            logger.error(
+                "\n{0} not found:"
+                "\n  Needed by: {1}"
+                "\n  Search path:\n    {2}".format(
+                    install_name, lib_fname, "\n    ".join(rpaths)
+                )
+            )
+            # At this point install_name is known to be a bad path.
+            # Expanding install_name with realpath may be undesirable.
+            yield realpath(install_name), install_name
+
+
+def walk_library(
+    lib_path,  # type: Text
+    filt_func=lambda filepath: True,  # type: Callable[[Text], bool]
+    visited=None,  # type: Optional[Set[Text]]
+):
+    # type: (...) -> Iterator[Text]
+    """
+    Yield all libraries on which `lib_path` depends, directly or indirectly.
+
+    First yields `lib_path` itself, if not already `visited` and then all
+    dependencies of `lib_path`, including dependencies of dependencies.
+
+    Dependencies which can not be resolved will be logged and ignored.
+
+    Parameters
+    ----------
+    lib_path : str
+        The library to start with.
+    filt_func : callable, optional
+        A callable which accepts filename as argument and returns True if we
+        should inspect the file or False otherwise.
+        Defaults to inspecting all files for library dependencies.
+        If `filt_func` filters a library it will also exclude all of that
+        libraries dependencies as well.
+    visited : None or set of str, optional
+        We update `visited` with new library_path's as we visit them, to
+        prevent infinite recursion and duplicates.  Input value of None
+        corresponds to the set `{lib_path}`.  Modified in-place.
+
+    Yields
+    ------
+    library_path : str
+        The paths of each library including `lib_path` without duplicates.
+    """
+    if visited is None:
+        visited = {lib_path}
+    elif lib_path in visited:
+        return
+    else:
+        visited.add(lib_path)
+    if not filt_func(lib_path):
+        logger.debug("Ignoring %s and its dependencies.", lib_path)
+        return
+    yield lib_path
+    for dependency_path, _ in get_dependencies(lib_path):
+        if not os.path.isfile(dependency_path):
+            logger.error(
+                "%s not found, requested by %s", dependency_path, lib_path,
+            )
+            continue
+        for sub_dependency in walk_library(
+            dependency_path,
```

Comment:
Just for consistency.

```suggestion
            dependency_fname,
```